nuclear_simulator/data_gen/optimization/timing_optimizer.py

```
@@ -0,0 +1,381 @@
+            return base_config, None, 0
+        
+        # Test baseline performance
+        baseline_trigger_time = self._test_trigger_timing(base_config, target_action, target_trigger_hours * 2)
+        
+        if self.verbose:
+            print(f"   📊 Baseline trigger time: {baseline_trigger_time:.3f}h" if baseline_trigger_time else "   📊 Baseline: no trigger detected")
+        
+        # If baseline is already within tolerance, return it
+        if baseline_trigger_time and abs(baseline_trigger_time - target_trigger_hours) <= tolerance_hours:
+            if self.verbose:
+                print(f"   ✅ Baseline already within tolerance!")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)